atintegrators/wiggler.m

```diff
@@ -20,7 +20,7 @@
 %  See also atdrift, atquadrupole, atsextupole, atsbend, atskewquad,
 %          atmultipole, atthinmultipole, atmarker, atcorrector
 
-ElemData = atwiggler(fname, Ltot, Lw, Bmax, Nstep, Nmeth, By, Bx, method);
+ElemData = atwiggler(fname, Ltot, Lw, Bmax, getenergy, method,'Nstep',Nstep, 'Nmeth',Nmeth, 'By',By, 'Bx', Bx);
 

```

Comment:
I cannot find `getenergy`. Am I missing a function?

Reply:
Sorry, I thought it was an AT function, which is the standard way to get the energy?

Reply:
Good question !
For a given lattice, you can get the energy with `atenergy(ring)`. It looks in order for:

- a `RingParam` element (new-style lattices)
- a `Cavity` element
- the global variable `GLOBVAL.E0`
- an `Energy` field in any element of the ring

Without a ring, you are left with only the global GLOBVAL.E0, which is not guaranteed to exist…
I am not very pleased with the present wiggler definition, it would have been better to describe it with a Bmax/(B.ro) normalised field, but that's how it is!
`atradon` will force correctly the energy in all radiating elements, including wigglers, but you may want to track a wiggler without radiation. That's why `atwiggler` needs explicitly the energy (though if defaults to zero if not provided).

So no obvious solution…

Reply:
Ok, I added the possibility to pass the energy as an extra argument in wiggler.m, if it is not there it will look at GLOBVAL.0, it it is not there it will get a default value 0. What do you think about it?

Reply:
I don't see anything better !